lib/utils.js

```diff
@@ -96,7 +96,15 @@ var Utils = module.exports = {
   cloneDeep: function(obj, fn) {
     return lodash.cloneDeep(obj, function (elem) {
       // Preserve special data-types like `fn` across clones. _.get() is used for checking up the prototype chain
-      if (elem && typeof elem.clone === 'function') {return elem.clone(); }
+      if (elem) {
+        // Allow objects to implement their own clone method
+        if (typeof elem.clone === 'function') {
+          // However do not mistakenly call clone methods on arrays and plain objects
+          if (!Array.isArray(elem) && !lodash.isPlainObject(elem)) {
+            return elem.clone();
+          }
+        }
+      }
```

Comment:
What i meant was go back to the old style and then add a short circuit at the top like:

```
if (Array.isArray(elem) || lodash.isPlainObject(elem)) return undefined;
```